src/main.ts

```diff
@@ -115,6 +115,49 @@ export type { PlatformInfo } from "./entities/platform-info";
 
 const ANONYMOUS_PREFIX = "$RCAnonymousID:";
 
+/**
+ * Configuration object for initializing the Purchases SDK.
+ *
+ * @example
+ * ```typescript
+ * // Object-based configuration (recommended)
+ * const purchases = Purchases.configure({
+ *   apiKey: "your_api_key",
+ *   appUserId: "user_123",
+ *   httpConfig: { additionalHeaders: { "Custom-Header": "value" } },
+ *   flags: { autoCollectUTMAsMetadata: true }
+ * });
+ *
+ * // Legacy separate arguments (deprecated)
+ * const purchases = Purchases.configure(
+ *   "your_api_key",
+ *   "user_123",
+ *   { additionalHeaders: { "Custom-Header": "value" } },
+ *   { autoCollectUTMAsMetadata: true }
+ * );
+ * ```
+ *
+ * @public
+ */
+export interface PurchasesConfig {
```

Comment:
Should we extract this to a separate file? Mostly so it doesn't grow the `main.ts` even more... If we do end up extracting it, we should make sure to export it from `main.ts` though, so it's available by consumers.